railties/lib/rails/application.rb

```diff
@@ -368,6 +367,19 @@ def config=(configuration) #:nodoc:
       @config = configuration
     end
 
+    def sandbox(&block)
+      config.sandbox.inject(block) {|stack, klass|
+        -> do
+          begin
+            klass.start_sandbox
+            stack.call
+          ensure
+            klass.finish_sandbox
+          end
+        end
+      }.call
+    end
+
```

Comment:
`sandbox` here is at application level (Rails.application.sandbox { ... }). Shouldn't it be a simple hook for libraries to use?

I remember you've mentioned file storage and message systems as examples that could use this hook, just as ActiveRecord does.

And if we use `connection_handler.connection_pools` to start sandbox on AR, then why would we need to specify AR classes?